pypeit/wavetilts.py

```diff
@@ -161,40 +212,32 @@ def __init__(self, mstilt, slits, spectrograph, par, wavepar, det=1, qa_path=Non
         # the slits
-        if self.slits is not None and self.mstilt is not None:
-            # NOTE: This uses the internal definition of `pad`
-            self.slitmask_science = self.slits.slit_img()
-            gpm = (self.mstilt.bpm == 0) if self.mstilt.bpm is not None \
-                                        else np.ones_like(self.slitmask_science, dtype=bool)
-            self.shape_science = self.slitmask_science.shape
-            self.shape_arc = self.mstilt.image.shape
-            self.nslits = self.slits.nslits
-            self.slit_left = arc.resize_slits2arc(self.shape_arc, self.shape_science,
-                                                  self.slits.left)
-            self.slit_righ = arc.resize_slits2arc(self.shape_arc, self.shape_science,
-                                                  self.slits.right)
-            self.slitcen   = arc.resize_slits2arc(self.shape_arc, self.shape_science,
-                                                  self.slits.center)
-            self.slitmask  = arc.resize_mask2arc(self.shape_arc, self.slitmask_science)
-            self.gpm = (arc.resize_mask2arc(self.shape_arc, gpm)) & (self.mstilt.image < self.nonlinear_counts)
-        else:
-            self.slitmask_science = None
-            self.shape_science = None
-            self.shape_arc = None
-            self.nslits = 0
-            self.slit_left = None
-            self.slit_righ = None
-            self.slitcen = None
-            self.slitmask = None
-            self.gpm = None
-        # --------------------------------------------------------------
+
+        # TODO -- Tidy this up into one or two methods?
+        # Load up all slits
+        # TODO -- Discuss further with JFH
+        all_left, all_right, mask = self.slits.select_edges(initial=True, flexure=self.spat_flexure)  # Grabs all, initial slits
+        self.tilt_bpm = np.invert(mask == 0)
+        self.tilt_bpm_init = self.tilt_bpm.copy()
+        # Slitmask
+        # TODO -- Discuss further with JFH
+        self.slitmask_science = self.slits.slit_img(initial=True, flexure=self.spat_flexure)  # All unmasked slits
+        # Resize
+        gpm = (self.mstilt.bpm == 0) if self.mstilt.bpm is not None \
+            else np.ones_like(self.slitmask_science, dtype=bool)
+        self.shape_science = self.slitmask_science.shape
+        self.shape_tilt = self.mstilt.image.shape
+        self.slitcen = arc.resize_slits2arc(self.shape_tilt, self.shape_science, (all_left+all_right)/2)
+        self.slitmask = arc.resize_mask2arc(self.shape_tilt, self.slitmask_science)
+        self.gpm = (arc.resize_mask2arc(self.shape_tilt, gpm)) & (self.mstilt.image < self.nonlinear_counts)
+    # --------------------------------------------------------------
 
         # Key Internals
         self.mask = None
-        self.all_trace_dict = [None]*self.nslits
+        self.all_trace_dict = [None]*self.slits.nslits
```

Comment:
Is this the total slits or total unmasked slits?

Reply:
Always total